File: Src/Icm.Core/Functions/TotalOrder/LongTotalOrder.vb
Namespace Icm
    Class LongTotalOrder
        Inherits BaseTotalOrder(Of Long)

        Public Overrides Function Least() As Long
            Return Long.MinValue
        End Function

        Public Overrides Function Long2T(ByVal d As Long) As Long
            Return d
        End Function

        Public Overrides Function Greatest() As Long
            Return Long.MaxValue
        End Function

        Public Overrides Function T2Long(ByVal t As Long) As Long
            Return t
        End Function

    End Class

End Namespace

File: Src/Icm.Core/Functions/TotalOrder/LongTotalOrder.cs
namespace Icm
{
	public class LongTotalOrder : BaseTotalOrder<long>
	{

		public override long Least()
		{
			return long.MinValue;
		}

		public override long Long2T(long d)
		{
			return d;
		}

		public override long Greatest()
		{
			return long.MaxValue;
		}

		public override long T2Long(long t)
		{
			return t;
		}

	}

}

//=======================================================
//Service provided by Telerik (www.telerik.com)
//Conversion powered by NRefactory.
//Twitter: @telerik
//Facebook: facebook.com/telerik
//=======================================================
